src/libmongoc/tests/json-test-operations.c

```diff
@@ -2255,6 +2255,49 @@ assert_event_count (json_test_ctx_t *ctx, const bson_t *operation)
    }
 }
 
+static void
+assert_session_transaction_state (json_test_ctx_t *ctx, const bson_t *operation)
+{
+   const char *expected_state;
+   const mongoc_client_session_t *session;
+
+   expected_state = bson_lookup_utf8 (operation, "arguments.state");
+   session = session_from_name (
+      ctx, bson_lookup_utf8 (operation, "arguments.session"));
+
+   if (!strcmp (expected_state, "none")) {
+      if (session->txn.state != MONGOC_INTERNAL_TRANSACTION_NONE) {
```

Comment:
Here and below. I'd used `mongoc_client_session_get_transaction_state` instead of reaching into the session to handle the case where `state == MONGOC_INTERNAL_TRANSACTION_COMMITTED_EMPTY`

Reply:
Sounds good! Used `mongoc_client_session_get_transaction_state` instead. Added `state_strings` to print actual names of states, too.

I could have used the switch on the transaction state, but since `assertSessionTransactionState.state` represents a subset of possible transaction states, I thought the else if ladder was more readable.